jupyter_kernel_mgmt/subproc/launcher.py

```diff
@@ -7,23 +7,45 @@
 import json
 import os
 import re
-import six
 import socket
 import stat
-from subprocess import PIPE
+from subprocess import PIPE, Popen
 import sys
 from traitlets.log import get_logger as get_app_logger
 import warnings
 
-from ipython_genutils.encoding import getdefaultencoding
 from jupyter_core.paths import jupyter_runtime_dir, secure_write
 from jupyter_core.utils import ensure_dir_exists
 from ..localinterfaces import localhost, is_local_ip, local_ips
-from .manager import KernelManager
+from .manager import KernelManager, use_sync_subprocess
 from ..util import run_sync
 
-port_names = ['shell_port', 'iopub_port', 'stdin_port', 'control_port',
-              'hb_port']
+port_names = ['shell_port', 'iopub_port', 'stdin_port', 'control_port', 'hb_port']
+
+if sys.platform.startswith("win") and sys.version_info >= (3, 8):
+    # Windows specific event-loop policy.  Although WindowsSelectorEventLoop is the current
+    # default event loop priot to Python 3.8, WindowsProactorEventLoop becomes the default
+    # in Python 3.8.  However, using WindowsProactorEventLoop fails during creation of
+    # IOLoopKernelClient because it doesn't implement add_reader(), while using
+    # WindowsSelectorEventLoop fails during asyncio.create_subprocess_exec() because it
+    # doesn't implement _make_subprocess_transport().  As a result, we need to force the use of
+    # the WindowsSelectorEventLoop, and essentially make process management synchronous on Windows
+    # (via use_sync_subprocess). (sigh)
+    # See https://github.com/takluyver/jupyter_kernel_mgmt/issues/31
+    # The following approach to this is from https://github.com/jupyter/notebook/pull/5047 by @minrk
+    try:
+        from asyncio import (
+            WindowsProactorEventLoopPolicy,
+            WindowsSelectorEventLoopPolicy,
+        )
+    except ImportError:
+        pass
+        # not affected
+    else:
+        if type(asyncio.get_event_loop_policy()) is WindowsProactorEventLoopPolicy:
+            # WindowsProactorEventLoopPolicy is not compatible with tornado 6
+            # fallback to the pre-3.8 default of Selector
+            asyncio.set_event_loop_policy(WindowsSelectorEventLoopPolicy())
```

Comment:
I'm not wild on setting this on import. Generally, I think it should be up to application code, not library code, to set global config like this.

Reply:
(Application code or test code, of course)

Reply:
Yeah, I was going to talk to you about this since it's potentially too far down stream in most cases.  I'll move it into the util directory as a method - where tests and more focused (i.e., thiner than server) applications can leverage its existence.